libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java

```diff
@@ -754,6 +757,20 @@ public void takeScreenshot(NavigationSnapshotReadyCallback navigationSnapshotRea
     mapboxMap.snapshot(navigationSnapshotReadyCallback);
   }
 
+  public void enableVanishingRouteLine() {
```

Comment:
why do you need **public** methods `enableVanishingRouteLine`/`disableVanishingRouteLine`? may be keep this logic private and invoke via lifecycle callbacks(`onStart`/`onStop`) only?

Reply:
When using NavigationView the NavigationMapboxMap is instantiated before the developer can indicate whether or not the vanishing route line feature can be enabled.  The only opportunity to enable the feature is through the NavigationViewOptions which are supplied when calling NavigationView.startNavigation().

Reply:
`onStart` lifecycle callback will be reached even if you attach lifecycleObserver after `onStart` of a LifecycleOwner(Activity/Fragment). So we're good to make `enableVanishingRouteLine` and `disableVanishingRouteLine` **private**

Reply:
The feature is disabled by default. How would you enable the feature from NavigationView if those methods are private?

Reply:
yeah, you're right. I thought it should work all the time, but now I see that it's enabled/disabled 👍